src/com/facebook/buck/apple/AppleLibraryDescription.java

```diff
@@ -771,21 +697,13 @@ private HeaderSymlinkTree createUnderlyingModuleSymlinkTreeBuildRule(
           Joiner.on(", ").join(buildTarget.getFlavors()));
       ImmutableSet.Builder<SourcePath> sourcePaths = ImmutableSet.builder();
       for (BuildTarget dep : args.getDeps()) {
-        Optional<FrameworkDependencies> frameworks =
-            graphBuilder.requireMetadata(
+        graphBuilder
+            .requireMetadata(
                 dep.withAppendedFlavors(
-                    AppleDescriptions.FRAMEWORK_FLAVOR,
-                    AppleDescriptions.NO_INCLUDE_FRAMEWORKS_FLAVOR,
-                    cxxPlatformFlavor.get()),
-                FrameworkDependencies.class);
-        if (frameworks.isPresent()) {
-          sourcePaths.addAll(frameworks.get().getSourcePaths());
-        }
+                    AppleDescriptions.NO_INCLUDE_FRAMEWORKS_FLAVOR, cxxPlatformFlavor.get()),
+                FrameworkDependencies.class)
+            .ifPresent(s -> sourcePaths.addAll(s.getSourcePaths()));
       }
-      // Not all parts of Buck use require yet, so require the rule here so it's available in the
-      // graphBuilder for the parts that don't.
-      BuildRule buildRule = graphBuilder.requireRule(buildTarget);
```

Comment:
@robbertvanginkel If we keep the `if` branch, why did we remove this code?

Reply:
Because the apple library won't be built as a framework, so it shouldn't add itself as a framework dependency.